pkg/proxmox/instances.go

```diff
@@ -131,85 +164,78 @@ func (i *instances) InstanceShutdown(ctx context.Context, node *v1.Node) (bool,
 func (i *instances) InstanceMetadata(ctx context.Context, node *v1.Node) (*cloudprovider.InstanceMetadata, error) {
 	klog.V(4).InfoS("instances.InstanceMetadata() called", "node", klog.KRef("", node.Name))
 
-	if providedIP, ok := node.ObjectMeta.Annotations[cloudproviderapi.AnnotationAlphaProvidedIPAddr]; ok {
-		var (
-			vmRef  *pxapi.VmRef
-			region string
-			err    error
-		)
```

Comment:
I would recommend making the changes this behavior in a separate PR.

Reply:
So in this case what I've done is made it so that the default is to continue with this logic - but when you switch to `auto` or any of the other non-default modes it won't use that logic. See the networking.md file that I created as it talks about the logic in detail.

Reply:
Yeah that's what it does. I explained more in my full comment that might help. We're both aiming for the same thing I think.